### ISSUE TITLE
CTRL+F should upgrade to CTRL+H when pressed

### ISSUE THREAD

**Issue by dylanh724** [notepad-plus-plus/notepad-plus-plus#10019]
## Currently
When on the CTRL+F window, pressing CTRL+H does nothing (or something unexpected).

## Expected
What _should_ intuitively happen like in most (if not all) other editors:

When on the CTRL+F window, **pressing CTRL+H should "upgrade" to what should normally show when pressing CTRL+H**.

## Debug
Notepad++ v7.9.5   (32-bit)
Build time : Mar 21 2021 - 02:09:07
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 2009
OS Build : 19043.1055
Current ANSI codepage : 1252
Plugins : JSLintNpp.dll JSMinNPP.dll mimeTools.dll NppConverter.dll NppExport.dll

**Comment by sasumner** [comment]
It really does seem like there should be a way to switch from one tab to another in the _Find_ family of window tabs, without using the mouse.  This is probably a duplicate issue but I haven't searched.

However, it probably _should not_ be Ctrl+f, Ctrl+h, etc., because these keys are remappable to other things.

The workaround at the moment is to use the mouse to switch tabs, or press Esc and then invoke the keycombo for the alternative function you want (e.g., viewing _Find_, press Esc, press Ctrl+h to view _Replace_).

**Comment by mere-human** [comment]
Another workaround is to use Shift+Tab to select the Find tab (it becomes focused):
![image](https://user-images.githubusercontent.com/9664141/124426933-65925b00-dd73-11eb-9c9f-445be6191c9a.png)
then use → key to select the Replace tab after Find:
![image](https://user-images.githubusercontent.com/9664141/124426980-7642d100-dd73-11eb-84bb-7ff741d7fb81.png)
In similar way, other tabs can be selected as well.

**Comment by sasumner** [comment]
>use Shift+Tab to select the Find tab...then use → key 

This works on any of the 5 tabs, as long as the _Find what_ has input focus when you do it.

Additionally, it refocuses _Find what_ after a switch of tab.
This makes moving from the _Find_ tab to the _Mark_ tab a rather large chore.

**Comment by donho** [comment]
>  it probably should not be Ctrl+f, Ctrl+h, etc., because these keys are remappable to other things.

What are they remapped for currently?

**Comment by sasumner** [comment]
>>it probably should not be Ctrl+f, Ctrl+h, etc., because these keys are remappable to other things.

>What are they remapped for currently?

Perhaps what I said earlier was misleading or confusing.  Let me try again:

So right now the _Find_ dialog has no idea what keycombo _Find_, _Replace_, etc. are mapped to in the _Shortcut Mapper_.

My thought initially was that the feature request would hardcode Ctrl+f press in the dialog to switch to _Find_ tab, Ctrl+h would be hardcoded to switch to _Replace_ tab, etc.  Thus if someone had remapped the editor window's replace keycombo to Ctrl+r via _Shortcut Mapper_, it made little sense to have them do Ctrl+h once in the _Find_ dialog in order to move to the _Replace_ tab.

However, the new code could be a bit more complicated and actually look up what the user has mapped in _Shortcut Mapper_ for find/replace/etc. and then use those same keycombos when the Find dialog has input focus.

But then I thought, Ctrl+Tab is sort of a standard for switching tabs in a tabbed window, so why not keep the code simpler and just use that to cycle through the tabs one by one until user stops on the one they want.

**Comment by donho** [comment]
> So right now the Find dialog has no idea what keycombo Find, Replace, etc. are mapped to in the Shortcut Mapper.

True

> But then I thought, Ctrl+Tab is sort of a standard for switching tabs in a tabbed window, so why not keep the code simpler and just use that to cycle through the tabs one by one until user stops on the one they want.

True also.  Yes we can  Ctrl+Tab totally

**Comment by donho** [comment]
This feature won't be implemented.

**Comment by mere-human** [comment]
@donho This is a very useful feature. It allows a user to use mouse less. I think it is especially useful for advanced users, who like shortcuts, plugins and stuff.

Also, some other editors support this more or less.
For example, in Visual Studio, when you open Find via _Ctrl + F_ then you may press _Ctrl + H_ and it will "upgrade" to Replace.

https://user-images.githubusercontent.com/9664141/125903100-14f09535-344e-4304-88b8-9c1bdb3b83ca.mp4

---

Could you, please, give this feature a try?

**Comment by dylanh724** [comment]
My philosophy is simple: If _Notepad_ has this feature, Notepad++ should have it.

**Comment by donho** [comment]
@dylanh724 
If you check the history closer, you will see the feature request was accepted, and it is implemented as you want exactly, and merged into the master:
https://github.com/notepad-plus-plus/notepad-plus-plus/pull/10177

**Comment by dylanh724** [comment]
> @dylanh724
> If you check the history closer, you will see the feature request was accepted, and it is implemented as you want exactly, and merged into the master:
> #10177

OH, gah -- misread from mobile. Misunderstanding, many thanks!
